length_field.go

```diff
@@ -1,13 +1,30 @@
 package sarama
 
-import "encoding/binary"
+import (
+	"encoding/binary"
+	"sync"
+)
 
 // LengthField implements the PushEncoder and PushDecoder interfaces for calculating 4-byte lengths.
 type lengthField struct {
 	startOffset int
 	length      int32
 }
 
+var lengthFieldPool = sync.Pool{}
```

Comment:
Can you please tell me more about use of pool, in current go runtime, any unreferenced sync.Pool objects are removed at each garbage collection cycle, so you can't use this as a long-lived objects. using a pool add another vector to memory leak for a long running process, want to make sure we are clear about our use here?

Reply:
Pool is perfectly fine to use for long running processes. As you mention, all pools are cleared on GC so no memory leak issues. This change means fewer allocations between GC cycles though. In my case, I conusme tens of millions of messages between GCs, so this reduces the number of allocated objects significantly. 

Note that pool behavior is [likely to be improved in 1.13](https://go-review.googlesource.com/c/go/+/166961/) for active pools.

Reply:
👍